### DISCUSSION TITLE
I just realized that isMissing are loaded from PhotoDB only once when osxphotos starts

### DISCUSSION THREAD

**Discussion by chengguangnan** [RhetTbull/osxphotos#1387] [General]
So it is possible that a photo that is marked as local would become missing when it's actually been exported. Right?  

For example, my Mac has about 50GB of free space on the system volumn and I'm exporting a 100GB album to an external disk. 

I see some "skipped missing photos" even using `--download-missing`. I think maybe what I mention above might be the cause? The photo was initially there but removed by Mac to save space during the export?

And I assume the solution with the current situation would be run the export command again until no more missing is reported? 

```
Skipping missing original photo IMG_6291.MOV (5D10D375-04CA-4014-8531-A109654018B2)
````

**Comment by RhetTbull** [comment]
Yes, the current version of osxphotos only loads the database once. The design of osxphotos means it reads the entire database and loads it into some python data structures in memory.  This means there's slow startup (and osxphotos cannot detect changes during the export) but faster in execution. Someday I plan to rewrite this part of osxphotos to dynamically query the database which will be slower but will be more robust and have faster startup and am working on that in a separate repo.  That's a long way off though.

In your case, yes, the best bet is probably to re-run osxphotos until all the missing photos have been downloaded and exported.
